src/App.tsx

```diff
@@ -1,24 +1,70 @@
 import React from 'react';
+import { Clock } from './Components/Clock';
+
 import './App.scss';
 
-const App: React.FC = () => {
-  const timerId: NodeJS.Timer = setInterval(() => {
-    const date: Date = new Date();
-
-    // eslint-disable-next-line
-    console.log(date.toLocaleTimeString());
-  }, 1000);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-      <p>
-        Current time:
-        {' '}
-        {/* Print the time here instead of DevTools */}
-      </p>
-    </div>
-  );
+type State = {
+  isClockVisible: boolean;
+  clockName: number;
 };
 
+class App extends React.Component<{}, State> {
+  state = {
+    isClockVisible: true,
+    clockName: 0,
+  };
+
+  clockVisibilityHandler = (isVisivle: boolean): void => {
+    this.setState({ isClockVisible: isVisivle });
+  };
+
+  createRandomName = () => {
+    if (this.state.isClockVisible) {
+      this.setState({
+        clockName: Math.round(Math.random() * 1000),
+      });
+    }
+  };
+
+  render() {
+    const { isClockVisible, clockName } = this.state;
+
+    return (
+      <div className="App">
+        <h1>React clock</h1>
+
+        {isClockVisible && <Clock name={clockName} />}
+
+        <button
+          className="App__button"
+          type="button"
+          onClick={() => this.clockVisibilityHandler(true)}
+        >
+          Show
+        </button>
+
+        {isClockVisible && (
+          <button
+            className="App__button"
+            type="button"
+            onClick={() => this.clockVisibilityHandler(false)}
+          >
+            Hide
+          </button>
+        )}
+
+        {isClockVisible && (
+          <button
+            className="App__button"
+            type="button"
+            onClick={() => this.createRandomName()}
```

Comment:
You can use shorter version if you don't pass any params to the function
```suggestion
            onClick={this.createRandomName}
```